### PR TITLE
Add support for GPT-4-O, "Omni" model

### PR DESCRIPTION
**Pull request by pkallos**

### CODE REVIEW

**Comment by pkallos** [comment]
@dqbd  seems the encoding model has changed, need to build/test support with [o200k_base](https://github.com/openai/tiktoken/commit/9d01e5670ff50eb74cdb96406c7f3d9add0ae2f8#diff-e5d4056b800e384f7fd41720372e64313c01e761fcd40e7cafe1ca558096e542R93)

**Comment by pkallos** [comment]
OK,  I have made the following changes in order to support the new encoding model:

- Copied the ranks source url, special token callouts, and PAT_STR regular expressions from the [openai/tiktoken project](https://github.com/openai/tiktoken/commit/9d01e5670ff50eb74cdb96406c7f3d9add0ae2f8#diff-e5d4056b800e384f7fd41720372e64313c01e761fcd40e7cafe1ca558096e542R91-R119)
- Added supports Rust code for constructors on the o200k_base encoder
- Adjusted the regex-rewriter in scripts/ranks.ts to support global replace of `?i:` patterns
- Added and adjusted a unit test for the new encoding in `wasm/test/test_simple_public.test.ts` which appears to work
- Pointed the `gpt-4o` model_to_encoding mappings to refer to the new `o200k_base` encoding

This has been tested as follows:
1. Remove the dep on wrangler in static/package.json using this patch
```patch
diff --git a/static/package.json b/static/package.json
index 8456899..f892d27 100644
--- a/static/package.json
+++ b/static/package.json
@@ -6,6 +6,5 @@
     "deploy": "wrangler pages publish ./src"
   },
   "devDependencies": {
-    "wrangler": "^1.15.0"
   }
 }
```
Unclear why but I always need to do this for `yarn install` to work, otherwise i get `➤ YN0001: │ Error: wrangler@npm:^1.15.0: No candidates found`
2. build the project with `turbo build`
3. run the tests with `turbo test`

File: scripts/ranks.ts [hunk 1]
```diff
@@ -264,8 +264,8 @@ async function main() {
       const bpe = await downloadBpe(data);
 
       if (lib === "js") {
-        bpe.pat_str = bpe.pat_str.replace(
-          /\(\?i:(.*?)\)/,
+        bpe.pat_str = bpe.pat_str.replaceAll(
+          /\(\?i:(.*?)\)/g,
```
Comment (pkallos) [inline scripts/ranks.ts:268]: It's necessary to do this as a global replace because `?i:` is now present several times in pat_str

**Comment by Pckool** [comment]
Please merge this 🙏

**Comment by dqbd** [comment]
Thank you so much @pkallos!

**Comment by gonzalonunez** [comment]
Amazing! Can we draft a new release?

**Comment by elliotmassen** [comment]
Hey @dqbd, do you know when version `1.0.15` might be published to NPM?

**Pull request merged**